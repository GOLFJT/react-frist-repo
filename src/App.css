.flex-box {
  display: flex;
  border-bottom:  1px solid #5E5E5E;
}
.col-flex{
  flex-grow: 1;
  padding: 20px;
}

.horizontal-flex {
  flex-direction: column;
}

.horizontal-flex .col-flex{
   border-bottom:  1px solid #5E5E5E;
}

.ta-style{
  width: 100%;
}


@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
