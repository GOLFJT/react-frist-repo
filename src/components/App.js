import React, { Component } from 'react'
import Post from './Post.js'
import { getApiSum } from './GetApi'
import { ThemeProvider } from 'styled-components'
import { 
  Section,
  darkTheme,
  lightTheme,
  InputTypePassword,
  Textarea,
  Button,
  Form,
  FormBox
} from './../styles/MainStyle'

class App extends Component {

  constructor(props) {
      super(props)
      this.state = {
           valTextarea: []
      }
  }  

  authFunction = () => {
      return true
  }

  deleteItem = (key) => {
    if(this.authFunction()){
      this.state.valTextarea.splice(key, 1)
      this.setState({
        valTextarea: this.state.valTextarea
      })
      return this.state.valTextarea
    }
  }

  click = (event) => {
    const inputText = event.target.main_input.value  
    if (inputText !== '') {
      let addPost = this.state.valTextarea
      addPost.push(inputText)
      this.setState({
         valTextarea: addPost
      })
      event.target.main_input.value = ''
    }
    return this.state.valTextarea
  }

  render() {
    return (
      <ThemeProvider theme={darkTheme}>
        <Section>
          <FormBox>
            <Form id="form-submit" onSubmit={(e) => {     
                    e.preventDefault() 
                    this.click(e)
                  } 
              }>
              <FormBox>
                  <Textarea rows="4" name="main_input" className="ta-style" id="inputText"></Textarea>
                  <InputTypePassword name="password"/>
              </FormBox>
              <FormBox>
                  <Button primary size={{ fontSize : "1.5em;" }} type="submit">Submit</Button>
              </FormBox>
            </Form >
          </FormBox>
          <Post 
            postItems={this.state.valTextarea} 
            deleteItem={ this.deleteItem }
          />
        </Section>  
      </ThemeProvider>
    )
  }
}

export default App
