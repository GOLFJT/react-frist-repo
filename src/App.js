import React, { Component } from 'react'
import './App.css'
import Post from './Post.js'
import { getApiSum } from './GetApi'

class App extends Component {

  constructor(props) {
      super(props)
      this.state = {
           valTextarea: []
      }
  }  

  authFunction = () => {
      return true
  }

  deleteItem = (key) => {
    if(this.authFunction()){
      this.state.valTextarea.splice(key, 1)
      this.setState({
        valTextarea: this.state.valTextarea
      })
      return this.state.valTextarea
    }
  }

  click = (event) => {
    const inputText = event.target.main_input.value  
    if (inputText !== '') {
      let addPost = this.state.valTextarea
      addPost.push(inputText)
      this.setState({
         valTextarea: addPost
      })
      event.target.main_input.value = ''
    }
    return this.state.valTextarea
  }

  render() {
    return (
      <div>
        <div className="flex-box">
          <form id="form-submit" onSubmit={(e) => {     
                  e.preventDefault() 
                  this.click(e)
                } 
            }>
            <div className="col-flex">
                <textarea rows="4" name="main_input" className="ta-style" id="inputText"></textarea>
                <input name="text" type="text"/>
            </div>
            <div className="col-flex">
                <button type="submit">Submit</button>
            </div>
          </form >
        </div>
        <Post 
          postItems={this.state.valTextarea} 
          deleteItem={ this.deleteItem }
        />
      </div>  
    )
  }
}

export default App
