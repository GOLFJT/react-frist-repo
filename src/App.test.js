import React from 'react'
import App from './App.js'
import renderer from 'react-test-renderer'
import { expect as chaiExpect } from 'chai'
import { shallow,mount } from 'enzyme'

describe('component app', () => {
  it('renders correctly app', () => {
  const jsonPost = renderer.create(<App/>).toJSON()
      expect(jsonPost).toMatchSnapshot()
  })

  it('id inputText heve 1', () => {
      const wrapper = shallow(<App/>)
      expect(wrapper.find('#inputText')).toHaveLength(1)
  })

  it('onSubmit call function Click', () => {
      const wrapper = mount(<App/>)
      const click = jest.spyOn(wrapper.instance(),'click')
      const fromSubmit = wrapper.find('#form-submit')
       //  const epreventDefault = () => {}

      // wrapper.find('#inputText').node.value = 'test'
      // fromSubmit.simulate('submit')
      fromSubmit.simulate('submit', {  target: { main_input: { value: 'test' } } } )
      expect(click).toHaveBeenCalledTimes(1)
  })

  it('test function click', () => {
      const resute = ['one-post','two-post','three-post']
      const wrapper = shallow(<App/>)
      wrapper.setState({ valTextarea: ['one-post','two-post'] })
      wrapper.instance().click({  target: { main_input: { value: 'three-post' } } })
      const newState = wrapper.state().valTextarea
      expect(newState).toEqual(resute)
  })

  it('test function deleteItem', () => {
      const resute = ['one-post','three-post']
      const wrapper = shallow(<App/>)
      wrapper.instance().authFunction = jest.fn().mockReturnValue(true)
      wrapper.setState({ valTextarea: ['one-post','two-post','three-post'] })
      wrapper.instance().deleteItem(1)
      const newState = wrapper.state().valTextarea
      expect(newState).toEqual(resute)
  })
})



