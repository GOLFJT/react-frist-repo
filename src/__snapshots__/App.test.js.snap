// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component app renders correctly app 1`] = `
<div>
  <div
    className="flex-box"
  >
    <form
      id="form-submit"
      onSubmit={[Function]}
    >
      <div
        className="col-flex"
      >
        <textarea
          className="ta-style"
          id="inputText"
          name="main_input"
          rows="4"
        />
        <input
          name="text"
          type="text"
        />
      </div>
      <div
        className="col-flex"
      >
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
  <div
    className="flex-box horizontal-flex"
  />
</div>
`;
